bin/drone-chef-supermarket

```diff
@@ -1,28 +1,63 @@
 #!/usr/bin/env ruby
 
+require "gli"
+
 $LOAD_PATH.push File.expand_path("../../lib", __FILE__)
 
 Signal.trap("INT") do
   abort
 end
 
 begin
-  require "drone"
   require "drone/supermarket"
 rescue LoadError
   require "rubygems"
-  require "drone"
   require "drone/supermarket"
 end
 
-ARGV.delete "--"
+include GLI::App
 
-begin
-  Drone::Plugin.new(ARGV.first) do |payload|
-    Drone::Supermarket.new(
-      payload
-    ).execute!
+def load_env(options, key)
+  if options[key.downcase.to_sym].nil? && !ENV["PLUGIN_#{key.upcase}"].nil?
+    options[key.downcase.to_sym] = ENV["PLUGIN_#{key.upcase}"]
+    options[key] = ENV["PLUGIN_#{key.upcase}"]
   end
-rescue Interrupt
-  abort "Quitting..."
 end
+
+def load_secrets(options, key)
+  if options[key.downcase.to_sym].nil? &&
+     !ENV["SUPERMARKET_#{key.upcase}"].nil?
+    options[key.downcase.to_sym] = ENV["SUPERMARKET_#{key.upcase}"]
+    options[key] = ENV["SUPERMARKET_#{key.upcase}"]
+  end
+end
+
+program_desc "Drone Supermarket Plugin"
+
+command :upload do |c|
+  c.flag [:u, :user]
+  c.flag [:k, :private_key], mask: true
+  c.flag [:s, :server]
+  c.flag [:v, :ssl_verify]
+  c.flag [:d, :debug]
+
+  c.action do |_global_options, options, _args|
+    %w(user server ssl_verify debug).each do |k|
```

Comment:
Just curious why you would do that.  `private_key` is a secret so why try to load from `PLUGIN_PRIVATE_KEY` and `SUPERMARKET_PRIVATE_KEY`?

Also, I think loading the secrets first would mean `PLUGIN_PRIVATE_KEY` would take priority if both were set.